### PR TITLE
Add Splunk Alerter functionnality with GUI configuration - With Plugin Config as Json stored in db

### PR DESCRIPTION
**Pull request by jynolen**
Reference of previous work #1009 
====
What have been change
 - Add a plugin system with auto-registration
 - Remove Splunk Alerter from Model 
 - Add Plugin Model to handle configuration
 - Add SplunkAlerter as plugin in this context
 - Add UnitTest for SplunkAlerter with local Splunk Collector Emulation

### CODE REVIEW

**Comment by jynolen** [comment]
FYI : I check the coverage of the critical code and it has beeen test well

File: env-config/config.py [hunk 1]
```diff
@@ -44,7 +44,11 @@
     'loggers': {
         'security_monkey': {
             'handlers': ['file', 'console'],
-            'level': 'DEBUG'
+            'level': 'INFO'
```
Comment (mikegrima) [inline env-config/config.py:47]: Please revert the changes to this file.

Comment (mikegrima) [reply env-config/config.py:47]: Also, can you add some docs on how to set up and configure this?

Comment (johnvox) [reply env-config/config.py:47]: All done, build in progress

File: docs/splunk_alerter.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+To enable it, add an ***inputs.conf*** file to your configuration
+Here an example of our configuration, don't worry  the token is a fake but match the pattern of real one
+
+    [http]
+    disabled = 0
+    enableSSL = 1
+    index = test
+    
+    [http://security_monkey]
+    token = ba903130-9c5f-4164-81fd-687714b02181
```
Comment (mikegrima) [inline docs/splunk_alerter.md]: Assuming this is sensitive. Please provide a warning to users about this.

File: security_monkey/common/utils.py [hunk 3]
```diff
@@ -80,6 +80,8 @@ def check_rfc_1918(cidr):
         return False
 
 
+
```
Comment (mikegrima) [inline security_monkey/common/utils.py]: Revert this.

File: security_monkey/datastore.py [hunk 4]
```diff
@@ -42,13 +42,15 @@
...
 from security_monkey.common.utils import sub_dict
+from security_monkey.exceptions import PluginConfigFormatError
 
 import datetime
 import json
 import hashlib
 import traceback
 
 
+
```
Comment (mikegrima) [inline security_monkey/datastore.py]: Delete this line.

File: security_monkey/datastore.py [hunk 5]
```diff
@@ -514,6 +516,19 @@ class WatcherConfig(db.Model):
     active = Column(Boolean(), nullable=False)
 
 
+class SMPluginConfig(db.Model):
+    __tablename__ = "sm_plugin_config"
+    id = Column(Integer, primary_key=True)
+    classname = Column(db.String(200), nullable=False)
```
Comment (mikegrima) [inline security_monkey/datastore.py]: Please make this a unique field.

File: security_monkey/plugins/splunk_alerter/http_event_collector.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
+"""
+    This class have been wrote by George Starcher <george@georgestarcher.com>
```
Comment (mikegrima) [inline security_monkey/plugins/splunk_alerter/http_event_collector.py]: Change to "was written".

**Comment by mikegrima** [comment]
One other thing I forgot to mention... please include the DB migration script to support the newly created table.

**Comment by johnvox** [comment]
Requested changes have been perform

**Comment by mikegrima** [comment]
Hello @jynolen : You will need to include the alembic DB migration code for the DB migration to be performed.

This is performed by running the `monkey db migrate` command to create a new DB revision, and modifying the file to ensure that it performs the proper DB migration code.

Also, please make the aforementioned changes to the `config.py` files.
